common/thorhelper/thorread.cpp

```diff
@@ -444,108 +602,414 @@ const void *BinaryDiskRowReader::nextRow()
-        //Multiple eogs should also be harmless if the engines switch to this representation.
-        if (eogPending)
+    return inlineNextRow([this,&resultSize](size32_t sizeRead, const byte * next){
+        if (needToTranslate)
         {
-            eogPending = false;
-            return eogRow;
+            //MORE: optimize the case where fields are lost off the end, and not bother translating - but return the modified size.
+            tempOutputBuffer.clear();
+            resultSize = translator->translate(bufferBuilder, *this, next);
+            const void * ret = bufferBuilder.getSelf();
+            bufferBuilder.finishRow(resultSize);
+            return ret;
         }
+        else
+        {
+            resultSize = sizeRead;
+            return (const void *)next;
+        }
+    });
+}
 
-        inputBuffer.finishedRow();
-        if (inputBuffer.eos())
-            return eofRow;
+//return a row allocated within a MemoryBufferBuilder
+const void *BinaryDiskRowReader::nextRow(MemoryBufferBuilder & builder)
+{
+    return inlineNextRow([this,&builder](size32_t sizeRead, const byte * next){
+        //MORE: optimize the case where fields are lost off the end, and not bother translating - but return the modified size.
+        if (needToTranslate)
+        {
+            size32_t resultSize = translator->translate(builder, *this, next);
+            const void * ret = builder.getSelf();
+            builder.finishRow(resultSize);
+            return ret;
+        }
+        else
+        {
+            builder.appendBytes(sizeRead, next);
+            return (const void *)(builder.getSelf() - sizeRead);
+        }
+    });
+}
 
-        //Currently each row in a stranded file contains a flag to indicate if the next is an end of strand.
-        //Is there a better way storing this (and combining it with the eog markers)?
-        if (stranded)
+
+
+
+//Common to IAllocRowStream and IRawRowStream
+bool BinaryDiskRowReader::getCursor(MemoryBuffer & cursor)
+{
+    //Is the following needed?
+    inputBuffer.finishedRow();
+
+    cursor.append(inputBuffer.tell());
+    cursor.append(eogPending);
+    return true;
+}
+
+void BinaryDiskRowReader::setCursor(MemoryBuffer & cursor)
+{
+    unsigned __int64 startPos;
+    cursor.read(startPos);
+    cursor.read(eogPending);
+
+    if (inputBuffer.tell() != startPos)
+        inputBuffer.reset(startPos);
+}
+
+void BinaryDiskRowReader::stop()
+{
+}
+
+
+// IDiskRowReader
+
+size32_t BinaryDiskRowReader::getFixedDiskRecordSize()
+{
+    size32_t fixedDiskRecordSize = actualDiskMeta->getFixedSize();
+    if (fixedDiskRecordSize && grouped)
+        fixedDiskRecordSize += 1;
+    return fixedDiskRecordSize;
+}
+
+
+//---------------------------------------------------------------------------------------------------------------------
+
+/*
+ * base class for reading a non-binary local file that uses IDynamicFieldValueFetcher to extract the values from
```

Comment:
Is there a reason the external file format would have to be non-binary? IE if I wanted to add parquet support would this be the base class I would use?